nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPath.java

```diff
@@ -255,6 +259,78 @@ public static AccessPath getAccessPathForNodeWithMapGet(Node node, @Nullable Typ
     }
   }
 
+  /**
+   * Constructs an access path ending with the class field element in the argument. The receiver is
+   * the method receiver itself.
+   *
+   * @param element the receiver element.
+   * @return access path representing the class field
+   */
+  public static AccessPath fromElement(Element element) {
+    assert element.getKind().isField()
+        : "element must be of type: FIELD but received: " + element.getKind();
+    Root root = new Root();
+    return new AccessPath(root, Collections.singletonList(new AccessPathElement(element)));
+  }
+
+  /**
+   * Extends an access path with a class field element.
+   *
+   * @param receiverNode the receiver node
+   * @param field element of the class field
+   * @return the extended access path
+   */
+  public static @Nullable AccessPath extendReceiverNodeAccessPathWithField(
+      @Nonnull Node receiverNode, Element field) {
+    if (receiverNode.getTree() == null) {
```

Comment:
Rather than a null check, can we check for a more specific type of `Node` that represents `this`?

Reply:
Actually the main intention here was to call ```fromElement``` for any ```null``` parameter (since the result of ```receiverNode.getTree()``` will be passed to ```extendReceiverTreeAccessPathWithField``` and it makes no sense to pass ```null``` there), not necessarily a ```this``` node. Do you still think I should change it ?